### PR TITLE
docs: add a link to next article

### PR DESCRIPTION
**Pull request by chenxsan**
## Changes

- Add a link to next article
- Add a link to previous article

While I'm reading the docs, I have to move my mouse to the sidebar all the time to click the next one while on mobile I have to click the burger menu every time. Adding a link to the next article at the end would save me a lot of time. Let me know what you think.

image

## Testing

I don't find any tests under docs, so I'm assuming it' not needed.

## Docs
Nothing.

### CODE REVIEW

**Comment by chenxsan** [comment]
Interesting, the links are missing on the preview url, that's unexpected when I expect the build to be same as dev environment. Any idea?

File: docs/src/components/PageContent/PageContent.astro [hunk 1]
```diff
@@ -38,4 +45,9 @@ import TableOfContents from '../RightSidebar/TableOfContents.tsx';
     <nav class="block sm:hidden">
         <MoreMenu editHref={githubEditUrl}/>
     </nav>
+    {
+      next && <aside>
+      Next Article: <a href={`${Astro.site.pathname}${next.link}`}>{next.text}</a>
```
Comment (chenxsan) [inline docs/src/components/PageContent/PageContent.astro:50]: I18n might be needed? Not sure about this as there's no i18n function available as far as I can see.

**Comment by jasikpark** [comment]
I love this idea! I'd say it'd also make sense to add a "Previous" link and also add "prev" and "next roles to the links!

**Comment by jasikpark** [comment]
like in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-rel

**Comment by chenxsan** [comment]
@jasikpark I've updated the pull request per your suggestion. Currently no styles are applied to the links, feel free to add it as you wish.

**Comment by jasikpark** [comment]
Oh! Also I think you can use `URL` to construct the path: `new URL(next.link, Astro.site)` which should work similarly, but 🤷‍♂️

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL might be good for making sure we don't accidentally have `docs.astro.build//hello-world`?

I think it'd be good to sort out the styles in this PR, not too sure about what they should be though, thanks for the good idea though!

**Comment by jasikpark** [comment]
On second thought, the styles you have now are fine for merging, they can be ✨ fancified ✨ in another PR if you don't have any immediate grand plans!

**Comment by chenxsan** [comment]
I've updated the `href` to use `new URL` syntax. However I'm using the `new URL().pathname` instead of the whole `new URL()` which would generate urls like `docs.astro.build/hello-world` for those previous/next links which makes it difficult to test when running `dev` locally.

**Comment by chenxsan** [comment]
We have this style on webpack documentation:



But I'm not content with it. So I'll leave the style in this pr as it is, maybe someone can comes up with a better one someday :)

**Comment by jasikpark** [comment]
> I've updated the `href` to use `new URL` syntax. However I'm using the `new URL().pathname` instead of the whole `new URL()` which would generate urls like `docs.astro.build/hello-world` for those previous/next links which makes it difficult to test when running `dev` locally.

Astro.site should resolve to localhost in dev mode, no?

**Comment by chenxsan** [comment]
Nope.



It's the same case for docs as `buildOptions` is set per https://github.com/snowpackjs/astro/blob/main/docs/astro.config.mjs#L3.

Could it be a bug?

**Comment by chenxsan** [comment]
Seems to be set on purpose according to https://github.com/snowpackjs/astro/blob/main/packages/astro/src/runtime.ts#L93.

**Comment by jasikpark** [comment]
Added an issue to track that, imo it's definitely a bug!

**Review by jasikpark — APPROVED** [review]

**Comment by jasikpark** [comment]
Merging with the plan to sort out styling at a later date!

**Pull request merged**